Add restrictions on Gateway APIs and services in the same namespace (#16597)
istio/istio.io: content/en/docs/tasks/traffic-management/ingress/gateway-api/index.md

To link a `Gateway` to a `Service`, configure the `addresses` field to point to a **single** `Hostname`.

{{< tip >}}

> Istio's controller does not configure the service if it's in a different namespace than the Gateway was created.

Reviewer: ```suggestion
Istio's controller will not configure the `Service` if it is in a different namespace to the `Gateway`.
```
